int16 mode
bool pos1
bool pos2
bool pos3